.github/workflows/bump-deps.yaml

```diff
@@ -0,0 +1,64 @@
+name: Bump DEPS
+on:
+  workflow_dispatch:
+    inputs:
+      dep:
+        description: "Dependency"
+        required: true
+        type: string
+        default: "vendor/web-discovery-project"
+      hash:
+        description: "New hash for dependency"
+        required: true
+        type: string
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-bump:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: auto-bump
+        run: |
+          set -x
+
+          git config user.email "${{ github.actor }}@users.noreply.github.com"
+          git config user.name "${{ github.actor }}"
+          git config push.default simple
+
+          SED=sed
+          DATE=date
+          DEPS=DEPS
+
+          # Inputs
+          DEP="${{ inputs.dep }}"
+          HASH="${{ inputs.hash }}"
+          [[ -z "$DEP" ]] && { echo Input dependency cannot be empty; exit 1; }
+          [[ -z "$HASH" ]] && { echo Input hash cannot be empty; exit 1; }
+
+          DATETIME="$($DATE '+%Y-%m-%d-%H-%M')"
+          readarray -t milestones < <(gh api /repos/brave/brave-core/milestones -q ".[] | .title" | grep -Fv Closed)
+          branches=( "${milestones[@]/*Nightly/master}" )
+          branches=( "${branches[@]/ -*/}" )
+          echo Branches to operate on:
+          declare -p branches
+          for branch in "${branches[@]}"
+          do
+            git switch "$branch"
+            pr_branch="pr-auto-bump-deps-${branch}-${DATETIME}"
+            git switch -c "$pr_branch"
+            "$SED" -i -e "s|\(.*${DEP}.*\)@\(.*\)\",$|\1@${HASH}\",|" "$DEPS"
+            git add ${DEPS}
+            git commit -m "Auto bump DEPS ${DEP}"
+            git push -u origin "$pr_branch"
+            gh pr create --base "${branch}" -f -t "Auto bump ${DEP}" -l "CI/run-audit-deps"
```

Comment:
Let's make this idempotent by checking for existing identical PRs before creating a new one.

People will trigger this more than once, we don't want to create duplicate PRs.

Reply:
We can leave this for a later iteration.